yarn-project/noir-protocol-circuits-types/src/artifacts/client/bundle.ts

```diff
@@ -30,12 +30,12 @@ export const SimulatedClientCircuitArtifacts: Record<ClientProtocolArtifact, Noi
 };
 
 export class BundleArtifactProvider implements ArtifactProvider {
-  getClientCircuitArtifactByName(artifact: ClientProtocolArtifact): Promise<NoirCompiledCircuit> {
-    return Promise.resolve(ClientCircuitArtifacts[artifact]);
+  getClientCircuitArtifactByName(artifact: ClientProtocolArtifact): Promise<NoirCompiledCircuitWithName> {
+    return Promise.resolve({ ...ClientCircuitArtifacts[artifact], name: artifact.replace('Artifact', '') });
```

Comment:
I introduced NoirCompiledCircuitWithName which as the name implies is just `NoirCompiledCircuit` type with the name. This was necessary as I needed the name in the circuit recorder.